File: haskell/isogram/src/Isogram.hs
module Isogram (isIsogram) where

import Data.List (length, map, filter, nub)
import Data.Char (isLower, toLower)

isIsogram :: String -> Bool
isIsogram sentence = length letters == length uniqueLetters
    where
        letters = filter isLower $ map toLower sentence 
        uniqueLetters = nub letters  

File: csharp/isogram/Isogram.cs
using System.Collections;
using System.Linq;

public static class Isogram
{
    public static bool IsIsogram(string word)
    {
        var seen = new BitArray(26);

        foreach (var c in word.Where(char.IsLetter).Select(char.ToLower))
        {
            if (seen.Get(c - 'a'))
                return false;

            seen.Set(c - 'a', true);
        }

        return true;
    }
}
